samples/OpenIDConnectSample/Startup.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.AspNet.Security.OpenIdConnect;
+using Microsoft.AspNet.Http.Security;
+using Microsoft.AspNet.Security;
+
+namespace OpenIdConnectSample
+{
+    public class Startup
+    {
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseServices(services =>
+            {
+                services.AddDataProtection();
+                services.Configure<ExternalAuthenticationOptions>(options =>
+                {
+                    options.SignInAsAuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
```

Comment:
To add some clarify, the code written in the tests under: OAuthBearerMiddlewareTests don't use metadata.  Inside any three of the notifications (message received, token received, token validated) you can take over.